lib/modules/manager/helmv3/common.ts

```diff
@@ -1,19 +1,40 @@
 import { quote } from 'shlex';
 import upath from 'upath';
 
+import { logger } from '../../../logger';
+import { coerceArray } from '../../../util/array';
 import type { ExtraEnv } from '../../../util/exec/types';
 import { privateCacheDir } from '../../../util/fs';
+import { addSecretForSanitizing } from '../../../util/sanitize';
+import { fromBase64 } from '../../../util/string';
+import { ecrRegex, getECRAuthToken } from '../../datasource/docker/ecr';
 import type { RepositoryRule } from './types';
 
-export function generateLoginCmd(
+export async function generateLoginCmd(
   repositoryRule: RepositoryRule,
   loginCMD: string
-): string | null {
-  const { username, password } = repositoryRule.hostRule;
-  if (username && password) {
+): Promise<string | null> {
+  const { hostRule, repository } = repositoryRule;
+  const { username, password } = hostRule;
+  if (ecrRegex.test(repository)) {
```

Comment:
```suggestion
  if (username !== 'AWS' && ecrRegex.test(repository)) {
```
check if we already have proper credentials

Reply:
I've made this change, but just wanted to highlight that the behaviour will now differ from the `docker` datasource, which doesn't seem to respect the `AWS` username:

https://github.com/renovatebot/renovate/blob/ba3c893a825755022638d03f7271b3e9c5b3f559/lib/modules/datasource/docker/common.ts#L92-L118

I also added another test to cover this case.